### ISSUE TITLE
Cannot configure severity for partial error code

### ISSUE THREAD

**Issue by Avasam** [microsoft/vscode-flake8#7]
I am trying to configure flake8 as such:
```py
class test:
    __init__(self): pass  # pylint: disable=syntax-error

    a = 0
    c = 2
    b = 1

    if True:
        if True:
            pass
```
```json
  "flake8.severity": {
    "convention": "Warning",
    "error": "Error",
    "fatal": "Error",
    "refactor": "Warning",
    "warning": "Warning",
    "info": "Warning",
    // builtins
    "A": "Warning",
    // mccabe
    "C": "Warning",
    // class attributes order
    "CCE": "Warning",
    // pycodestyles
    "E": "Warning",
    "E9": "Error", // Runtime
    "W": "Warning",
    // Pyflakes
    "F": "Warning",
    // PEP8 Naming convention
    "N": "Warning",
    // Simplify
    "SIM": "Warning",
    "SIM9": "Information",
    // PYI
    "Y": "Warning",
  },
```
Only single characters or full error code work. The following do nothing: `CCE`, `E9`, `SIM`, `SIM9`.

**Comment by karthiknadig** [comment]
The check happens here, `code_type` is basically category. Pattern matching not supported:
https://github.com/microsoft/vscode-flake8/blob/42c7979e76dca95805a6fc6783854456abe252fc/bundled/tool/server.py#L108-L118

One of the reasons we don't allow full pattern matching, but only category or error code is this: let say you have `E`, `E1`, `E13` all set to different severity, you now have to build a hierarchy that ensures that you check for `E13` first, then `E1`, and then `E`, and before that you have to check for all absolute error codes.

**Comment by gschwaer** [comment]
It should be sufficient to iterate over the categories by length from longest to shortest and pick the first that fits, right?
